processor/src/test/java/org/mapstruct/ap/test/inheritfromconfig/CarMapperWithAutoInheritance.java

```diff
@@ -26,24 +25,24 @@
 import org.mapstruct.factory.Mappers;
 
 /**
- * @author Sjaak Derksen
+ * @author Andreas Gudian
+ *
  */
-@Mapper
-public interface SourceTargetMapperErroneouslyAnnotated2 {
-
-    SourceTargetMapperErroneouslyAnnotated2 INSTANCE
-            = Mappers.getMapper( SourceTargetMapperErroneouslyAnnotated2.class );
+@Mapper(
+    config = AutoInheritedConfig.class
+)
+public interface CarMapperWithAutoInheritance {
```

Comment:
Ah, I see. Is there a need for this restriction? I think there is no fundamental difference between the two cases: a) "prototype methods" declared on a config class or b) prototype methods declared on the mapper or it supertypes. So IMO we should handle them equally.